Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -161,48 +191,56 @@ extension ActivityStreamPanel {
             }
         }
 
-        func sectionInsets() -> CGFloat {
+        /*
+         There are edge cases to handle when calculating section insets
+        - An iPhone 7+ is considered regular width when in landscape
+        - An iPad in 66% split view is still considered regular width
+         */
+        func sectionInsets(_ traits: UITraitCollection, frameWidth: CGFloat) -> CGFloat {
+            var currentTraits = traits
+            if (traits.horizontalSizeClass == .regular && UIScreen.main.bounds.size.width != frameWidth) || UIDevice.current.userInterfaceIdiom == .phone {
+                currentTraits = UITraitCollection(horizontalSizeClass: .compact)
+            }
             switch self {
             case .highlights:
-                return UIDevice.current.userInterfaceIdiom == .pad ? ASPanelUX.SectionInsetsForIpad + ASHorizontalScrollCellUX.MinimumInsets : ASPanelUX.SectionInsetsForIphone
+                var insets = ASPanelUX.sectionInsetsForSizeClass[currentTraits.horizontalSizeClass]
+                insets = insets + ASPanelUX.MinimumInsets
+                return insets
             case .topSites:
-                return UIDevice.current.userInterfaceIdiom == .pad ? ASPanelUX.SectionInsetsForIpad : 0
+                return ASPanelUX.sectionInsetsForSizeClass[currentTraits.horizontalSizeClass]
             case .highlightIntro:
-                return UIDevice.current.userInterfaceIdiom == .pad ? ASPanelUX.SectionInsetsForIpad : 0
+                return ASPanelUX.sectionInsetsForSizeClass[currentTraits.horizontalSizeClass]
             }
         }
 
-        func numberOfItemsForRow() -> CGFloat {
+        func numberOfItemsForRow(_ traits: UITraitCollection) -> CGFloat {
             switch self {
             case .highlights:
-                var numItems: CGFloat = 0
-                if UIDevice.current.orientation == .landscapeLeft || UIDevice.current.orientation == .landscapeRight {
-                    numItems = 4
-                } else if UIDevice.current.userInterfaceIdiom == .pad {
-                    numItems = 3
-                } else {
-                    numItems = 2
+                var numItems: CGFloat = ASPanelUX.numberOfItemsPerRowForSizeClsssIpad[traits.horizontalSizeClass]
+                if UIInterfaceOrientationIsPortrait(UIApplication.shared.statusBarOrientation) {
```

Comment:
UIDeviceOrientation is different than the actual orientation of the user interface.
https://developer.apple.com/documentation/uikit/uidevice/1620053-orientation

From what I understand. Sometimes the orientation can be faceUp/faceDown while the UI is actually in portrait. This becomes hard to detect. Using the statusbar orientation ensures that the orientation of our view matches the status bar which is always in the correct orientation.